net/dns/resolver/forwarder.go

```diff
@@ -577,6 +623,89 @@ func (f *forwarder) sendUDP(ctx context.Context, fq *forwardQuery, rr resolverAn
 	return out, nil
 }
 
+func (f *forwarder) sendTCP(ctx context.Context, fq *forwardQuery, rr resolverAndDelay) (ret []byte, err error) {
+	ipp, ok := rr.name.IPPort()
+	if !ok {
+		metricDNSFwdErrorType.Add(1)
+		return nil, fmt.Errorf("unrecognized resolver type %q", rr.name.Addr)
+	}
+	metricDNSFwdTCP.Add(1)
+	ctx = sockstats.WithSockStats(ctx, sockstats.LabelDNSForwarderTCP, f.logf)
+
+	// Specify the exact family to work around https://github.com/golang/go/issues/52264
+	tcpFam := "tcp4"
+	if ipp.Addr().Is6() {
+		tcpFam = "tcp6"
+	}
+
+	conn, err := f.dialer.SystemDial(ctx, tcpFam, ipp.String())
```

Comment:
should we attempt to keep these around or are they a use once connection?

Reply:
They can be re-used, but some DNS servers will close the connection after a single query so I didn't really want to deal with detecting that, handling cache sizing/eviction, etc.

Reply:
How so? If we hit this codepath, we just sent a UDP packet and got a truncated response so the device radio should already be active. I'm not super familiar with how battery management works; is a TCP connection different than UDP?

Reply:
just thinking about the tcp handshake + retry loops